pkg/apis/pipeline/v1/pipeline_validation_test.go

```diff
@@ -1127,6 +1127,38 @@ func TestValidatePipelineParameterVariables_Success(t *testing.T) {
 				Name: "a-param-intended-to-be-array", Value: ParamValue{Type: ParamTypeString, StringVal: "$(params.myArray[*])"},
 			}},
 		}},
+	}, {
+		name: "valid array parameter variables in matrix",
```

Comment:
I'm confused what this is supposed to test-- it looks like it's testing valid string parameters in matrix.include (and also has some unrelated array params), but the test name sounds like it's testing array params in matrix.params. Can you update the test name and remove the unrelated array params?

Reply:
Updated error messaging